ffc/codegeneration/codegeneration.py

```diff
@@ -29,59 +32,54 @@ def generate_code(analysis, object_names, ir, parameters, jit):
 
     logger.debug("Compiler stage 4: Generating code")
 
-    full_ir = ir
-
     # FIXME: This has global side effects
     # Set code generation parameters
     # set_float_formatting(parameters["precision"])
     # set_exception_handling(parameters["convert_exceptions_to_warnings"])
 
-    # Extract data from analysis
-    form_data, elements, element_map, domains = analysis
-
-    # Extract representations
-    ir_finite_elements, ir_dofmaps, ir_coordinate_mappings, ir_integrals, ir_forms = ir
-
     # Generate code for finite_elements
-    logger.debug("Generating code for {} finite_element(s)".format(len(ir_finite_elements)))
+    logger.debug("Generating code for {} finite_element(s)".format(len(ir.elements)))
     code_finite_elements = [
-        ufc_finite_element_generator(ir, parameters) for ir in ir_finite_elements
+        ufc_finite_element_generator(ir, parameters) for ir in ir.elements
     ]
 
     # Generate code for dofmaps
-    logger.debug("Generating code for {} dofmap(s)".format(len(ir_dofmaps)))
-    code_dofmaps = [ufc_dofmap_generator(ir, parameters) for ir in ir_dofmaps]
+    logger.debug("Generating code for {} dofmap(s)".format(len(ir.dofmaps)))
+    code_dofmaps = [ufc_dofmap_generator(ir, parameters) for ir in ir.dofmaps]
 
     # Generate code for coordinate_mappings
-    logger.debug("Generating code for {} coordinate_mapping(s)".format(len(ir_coordinate_mappings)))
+    logger.debug("Generating code for {} coordinate_mapping(s)".format(len(ir.coordinate_mappings)))
     code_coordinate_mappings = [
-        ufc_coordinate_mapping_generator(ir, parameters) for ir in ir_coordinate_mappings
```

Comment:
This looks confusing "`for ir in ir.coordinate_mappings`". Maybe the iterator variable `ir` should be changed?